### PR TITLE
Modify draw interaction dispatch order

### PR DESCRIPTION
**Pull request by gberaudo**
When finishing drawing, this PR dispatches the `drawend` event before adding
the feature to the source or collection.

This change allows user code to finish configuring the feature (setting style,
properties, ...).

It is useful since inserting a feature, or later on modifying it, may trigger
processing which only makes sense if the feature is ready. For example,
ol3-cesium will recreate a 3D counterpart.

### CODE REVIEW

**Comment by fredj** [comment]
LGTM

**Comment by bartvde** [comment]
can we at least have a test case to check for the correct order?

**Comment by tsauerwein** [comment]
On the other hand there might be applications that rely on that the features are already in the source or collection when `drawend` is called. What about a new event `before:drawend`?

Anyway we should mention this in the documentation.

**Comment by fredj** [comment]
@tsauerwein How can the feature be already in the source/collection ?

**Comment by tsauerwein** [comment]
@fredj Before this change the new feature was added to the source and then the event was triggered. And my point is that there might be applications that rely on that behavior.

**Comment by gberaudo** [comment]
The method is experimental and did not promise anything about events.
Since there is no clear use case, I think it is better to keep the API simple
and not introduce another event.

I added documentation.

**Comment by gberaudo** [comment]
@bartvde, I added a test case as requested.
I also updated the changelog.
I think it is now ready for merging; is it OK @tsauerwein?

File: changelog/upgrade-notes.md [hunk 1]
```diff
@@ -118,7 +118,11 @@ size of a sub-rectangle in an image sprite.
 
 ### Support for non-square tiles
 
-The return value of ´ol.tilegrid.TileGrid#getTileSize()` will now be an `ol.Size` array instead of a number if non-square tiles (i.e. an `ol.Size` array instead of a number as `tilsSize`) are used. To always get an `ol.Size`, the new `ol.size.toSize()` was added.
+The return value of `ol.tilegrid.TileGrid#getTileSize()` will now be an `ol.Size` array instead of a number if non-square tiles (i.e. an `ol.Size` array instead of a number as `tilsSize`) are used. To always get an `ol.Size`, the new `ol.size.toSize()` was added.
+
+### Change to `ol.interaction.Draw`
+
+When finishing a draw, the `ol.DrawEventType.DRAWEND` event is now dispatched before the feature is inserted to either the source or the collection. This change allows application code to finish setting up the feature.
```
Comment (bartvde) [inline changelog/upgrade-notes.md]: should we use the string value here instead because that's what users will use?

**Comment by tsauerwein** [comment]
We still can add `before:drawend` or alike when there is a real use-case. Thanks for updating the changelog.

**Comment by bartvde** [comment]
thanks for adding the test case @gberaudo appreciated

File: changelog/upgrade-notes.md (hunk 1 above)
Comment (gberaudo) [inline changelog/upgrade-notes.md]: Right @bartvde, I just updated the commit with your suggestion.

**Comment by bartvde** [comment]
cool we can merge if Travis passes

**Comment by gberaudo** [comment]
Thank you for this thorough review.

**Pull request merged**